modules/light-clients/06-solomachine/types/client_state.go

```diff
@@ -39,6 +39,20 @@ func (cs ClientState) GetLatestHeight() exported.Height {
 	return clienttypes.NewHeight(0, cs.Sequence)
 }
 
+// GetTimestampAtHeight returns the timestamp in nanoseconds of the consensus state at the given height.
+func (cs ClientState) GetTimestampAtHeight(
+	_ sdk.Context,
+	clientStore sdk.KVStore,
+	cdc codec.BinaryCodec,
+	height exported.Height,
+) (uint64, error) {
+	consensusState, err := GetConsensusState(clientStore, cdc, height)
+	if err != nil {
+		return 0, sdkerrors.Wrapf(err, "height (%s)", height)
+	}
+	return consensusState.GetTimestamp(), nil
```

Comment:
solo machine client state has access to the consensus state, no need to call `GetConsensusState`. This is actually why timeouts don't work for solo machines (they don't store a consensus state for each height they have a consensus state). Only the latest consensus state is tracked/needed

Reply:
I added a check to see if `height` passed in matches latest height or return an error if that makes sense?